engine/src/test/java/com/datatorrent/stram/StramLocalClusterTest.java

```diff
@@ -395,5 +396,20 @@ private String generatejar(String pojoClassName) throws IOException, Interrupted
     return new File(destDir, pojoClassName + ".jar").getAbsolutePath();
   }
 
+  @Test
+  public void testAppPath() throws Exception
+  {
+    TestGeneratorInputOperator o1 = dag.addOperator("o1", TestGeneratorInputOperator.class);
+    o1.setMaxTuples(1);
+    File relPath = new File(dag.getAttributes().get(DAGContext.APPLICATION_PATH));
+    String uriPath = relPath.toURI().toString();
+    dag.setAttribute(DAGContext.APPLICATION_PATH, uriPath);
+    StramLocalCluster cluster = new StramLocalCluster(dag);
+    // no need for run()
+    cluster.shutdown();
```

Comment:
It wasn't doing what I wanted (shutting down the buffer server). I moved launch of the buffer server from constructor into the run method so there is no need for this any longer.